### PR TITLE
TextInput theme

### PR DESCRIPTION
**Pull request by andrewberg-okta**

### CODE REVIEW

File: packages/odyssey-react/src/components/TextInput/TextInput.module.scss [hunk 1]
```diff
@@ -11,10 +11,68 @@
...
+  }
+
+  &::placeholder {
+    color: var(--ColorTextPlaceholder);
+  }
 
   &[type="search"] {
-    padding-inline-start: ms(1) + $spacing-s + $spacing-xs;
+    padding-inline-start: calc(
+      var(--IndicatorSize) + var(--IndicatorInsetInlineStart) +
```
Comment (edburyenegren-okta) [inline packages/odyssey-react/src/components/TextInput/TextInput.module.scss:73]: 👏🏼 nice work untangling my unlabeled math!

File: packages/odyssey-react/src/components/TextInput/TextInput.module.scss [hunk 2]
```diff
@@ -29,7 +87,7 @@
     }
 
     [dir="rtl"] & {
-      background-position: right $spacing-s center;
+      background-position: right var(--RtlBackgroundPositionRightOffset) center;
```
Comment (edburyenegren-okta) [inline packages/odyssey-react/src/components/TextInput/TextInput.module.scss:90]: Huh. It's weird that we weren't specific about this for the `ltr` default. I expected this variable to be something like `BackgroundPositionOffset`, used in two places with the `background-position` property varying between `right` and `left` for RTL/LTR.

What am I missing about my own CSS? 😬

**Review by edburyenegren-okta — APPROVED** [review]
Looks great! I have one question about how we're declaring the RTL background-position, but that could be tackled in a separate refactor if needed.

File: packages/odyssey-react/src/components/TextInput/TextInput.module.scss [hunk 3]
```diff
@@ -11,10 +11,68 @@
...
 .root {
-  @include input-baseline;
+  @include border-radius(var(--BorderRadius));
+
+  display: block;
+  position: relative;
+  width: 100%;
+  max-width: var(--MaxWidth);
+  margin-block: 0;
+  margin-inline: 0;
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/src/components/TextInput/TextInput.module.scss:21]: these types of styles (margin/padding) were part of the request options to theme. We can update and set them up as component only variables so something like this:

```
// theme module
SpaceMarginBlock: 0,
SpaceMarginInline: 0,

// style module
margin-block: var(--SpaceMarginBlock);
margin-inline: var(--SpaceMarginInline);
```

File: packages/odyssey-react/src/components/TextInput/TextInput.module.scss [hunk 4]
```diff
@@ -11,10 +11,68 @@
...
+  border: var(--BorderWidth) var(--BorderStyle) var(--ColorBorder);
+  background-color: var(--ColorBackground);
+  color: var(--ColorText);
+  font-family: var(--FontFamily);
+  font-size: var(--FontSize);
+  line-height: var(--LineHeight);
+
+  &:hover,
+  &:focus {
+    border-color: cv("blue");
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/src/components/TextInput/TextInput.module.scss:36]: `var(--ColorBorderPrimaryBase)` should be equivalent here and allow for theming borders anotehr requirement from SIW team

**Review by jeffbelser-okta — COMMENTED** [review]
strong work - comments inline

File: packages/odyssey-react/src/components/TextInput/TextInput.theme.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  MaxWidth: theme.FontLineLengthMax,
+  FontFamily: theme.FontFamilyBase,
+  FontSize: theme.FontSizeBody,
+  LineHeight: theme.FontLineHeightBase,
+  FocusOutlineWidth: theme.FocusOutlineWidthBase,
+  TransitionDuration: theme.TransitionDurationBase,
+  TransitionTiming: theme.TransitionTimingBase,
+  RtlBackgroundPositionRightOffset: theme.SpaceRemS,
+  SearchPaddingInlineStart: theme.SpaceRemXs,
+  IndicatorSize: "1.1487rem",
```
Comment (edburyenegren-okta) [inline packages/odyssey-react/src/components/TextInput/TextInput.theme.ts:40]: `theme.FontSizeHeading5`? (Naming aside, I believe this is intended to be associated with the size one step up from the base of the scale.)

Comment (jeffbelser-okta) [reply packages/odyssey-react/src/components/TextInput/TextInput.theme.ts:40]: On topic: I encountered one size that wasn't part of our current mapping in Text but decided against adding a new entry (kept it local for now)

Comment (andrewberg-okta) [reply packages/odyssey-react/src/components/TextInput/TextInput.theme.ts:40]: Yes, the previous implementation was using the scale value to go up one step. Mapping this value back to something intended for heading sizes feels more brittle than a magic number right now.

File: packages/odyssey-react/src/components/TextInput/TextInput.module.scss (hunk 2 above)
Comment (andrewberg-okta) [reply packages/odyssey-react/src/components/TextInput/TextInput.module.scss:90]: Looking closer this whole rule set isn't used anymore and can be safely removed. It only applies to the previous implementation where the search icon was a background image.

File: packages/odyssey-react/src/components/TextInput/TextInput.module.scss (hunk 4 above)
Comment (andrewberg-okta) [reply packages/odyssey-react/src/components/TextInput/TextInput.module.scss:36]: 👍

**Review by jeffbelser-okta — APPROVED** [review]

File: packages/odyssey-react/src/components/TextInput/TextInput.theme.ts (hunk 5 above)
Comment (edburyenegren-okta) [reply packages/odyssey-react/src/components/TextInput/TextInput.theme.ts]: Good call out @andrewberg-okta. @josesolano-okta and I can revisit this when we propose the new type tokens.

I think we'll want to find a standard for these Icon/Indicator sizes that doesn't attach them to something as specific as a Heading style.

**Pull request merged**